File: First Unique Character in a String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> char_to_cnt;
        for (auto c : s)
            char_to_cnt[c]++;
        for (int i = 0; i < s.length(); i++)
            if (char_to_cnt[s[i]] == 1)
                return i;
        return -1;
    }
};

File: First Unique Character in a String.java
public class Solution {
    public int firstUniqChar(String s) {
        HashSet<Character> cset = new HashSet<Character>();
        LinkedHashMap<Character, Integer> char_to_idx = new LinkedHashMap<Character, Integer>();
        for (int i = 0; i < s.length(); i++){
            if (cset.contains(s.charAt(i)))
                char_to_idx.remove(s.charAt(i));
            else{
                cset.add(s.charAt(i));
                char_to_idx.put(s.charAt(i), i);
            }
        }
        Iterator<Map.Entry<Character, Integer>> it = char_to_idx.entrySet().iterator();
        return it.hasNext() ? it.next().getValue() : -1;
    }
}